### PR TITLE
Feature/UI refactor

### PR DESCRIPTION
**Pull request by antongotech**

### CODE REVIEW

File: src/pages/deployer/data.ts [hunk 1]
```diff
@@ -41,15 +41,15 @@ const onchainFormSpec = [
...
-    description: "Optional freeform sentence explaining about your project",
+    description: "Optional sentence explaining about your project.",
     type: "string",
     default: "Low fee peer-to-peer electronic cash alternative to Bitcoin",
   },
 
   {
     name: "tokenImage",
-    label: "Jetton Logo URI",
-    description: "URL of 256x256 pixel PNG image of token logo with transparent background",
```
Comment (shaharyakir) [inline src/pages/deployer/data.ts:52]: why did we remove "with transparent background"?

File: src/pages/deployer/index.tsx [hunk 2]
```diff
@@ -158,43 +213,69 @@ function Description() {
...
-        protocol. This free educational tool allows you to deploy your own Jetton to mainnet in one
-        click. You will need at least 0.25 TON for deployment fees. <br />
+        . This free educational tool allows you to deploy your own Jetton to mainnet in one click.
+        You will need at least 0.25 TON for deployment fees. <br />
         <Spacer />
         For detailed instructions and in-depth explanations of all fields please see the{" "}
-        <Link
-          target="_blank"
-          href="https://github.com/ton-defi-org/jetton-deployer-contracts#jetton-metadata-field-best-practices">
+        <Link target="_blank" href="https://github.com/ton-blockchain/minter-contract">
```
Comment (shaharyakir) [inline src/pages/deployer/index.tsx:220]: restore hash

File: src/pages/deployer/index.tsx [hunk 3]
```diff
@@ -158,43 +213,69 @@ function Description() {
...
           GitHub Pages
         </Link>
         . <Spacer />
-        Is this deployer safe? Yes! Read{" "}
-        <Link
-          target="_blank"
-          href="https://github.com/ton-defi-org/jetton-deployer-contracts#protect-yourself-and-your-users">
-          this
+        Is this deployer safe? Yes!{" "}
+        <Link target="_blank" href="https://github.com/ton-blockchain/minter-contract">
```
Comment (shaharyakir) [inline src/pages/deployer/index.tsx:240]: restore hash

File: src/pages/deployer/index.tsx [hunk 4]
```diff
@@ -158,43 +213,69 @@ function Description() {
...
+          Read
         </Link>{" "}
-        to understand why.
+        this to understand.
```
Comment (shaharyakir) [inline src/pages/deployer/index.tsx:243]: restore "why"

Comment (shaharyakir) [inline src/pages/deployer/index.tsx:243]: link should be around "this" instead of "read"

File: src/pages/jetton/context/JettonActionsContext.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import React, { createContext, useState } from "react";
+
+const JettonActionsContext = createContext({
```
Comment (shaharyakir) [inline src/pages/jetton/context/JettonActionsContext.tsx:3]: I think recoil would be better for this. contexts are usually inefficient for rendering if they're value is an object that changes (and also we just don't really use them as a standard). They're good as far as I'm aware for static/nearly-static properties such as user's language, theme etc.

So up to you whether to refactor, but let's avoid them altogether moving forward.

File: src/pages/deployer/index.tsx [hunk 6]
```diff
@@ -111,24 +108,68 @@ function DeployerPage() {
...
+                  padding: 3,
+                }}>
+                <Typography
+                  variant="h5"
+                  mb={3}
+                  sx={{
+                    color: "#161C28",
+                    fontSize: 20,
+                    fontWeight: 800,
+                  }}>
```
Comment (shaharyakir) [inline src/pages/deployer/index.tsx:163]: is it possible to extract styling out? it makes the component harder to read

File: src/pages/deployer/index.tsx [hunk 7]
```diff
@@ -158,43 +213,69 @@ function Description() {
...
+          }}>
+          <img
+            width={14}
+            height={14}
+            style={{ position: "absolute", top: "-6px" }}
+            src={rightArrow}
+            alt="Icon"
+          />
+        </Box>
+      </IconButton>
```
Comment (shaharyakir) [inline src/pages/deployer/index.tsx:278]: consider extracting to its own component

File: src/pages/jetton/actions/transfer/utils.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import { isValidAddress } from "utils";
+import BN from "bn.js";
+import BigNumberDisplay from "components/BigNumberDisplay";
+
+export const getError = (
```
Comment (shaharyakir) [inline src/pages/jetton/actions/transfer/utils.tsx:5]:
```suggestion
export const validateTransfer = (
```
nit

File: src/pages/jetton/util.ts [hunk 9]
```diff
@@ -1,16 +1,13 @@
...
 import { JettonDetailMessage } from "./types";
 export { BigNumber } from "bignumber.js";
 
-const commonGithubUrl =
-  "https://github.com/ton-defi-org/jetton-deployer-contracts#protect-yourself-and-your-users";
+const commonGithubUrl = "https://github.com/ton-blockchain/minter-contract";
 
-const offChainGithubUrl =
-  "https://github.com/ton-defi-org/jetton-deployer-contracts#jetton-metadata-field-best-practices";
+const offChainGithubUrl = "https://github.com/ton-blockchain/minter-contract";
```
Comment (shaharyakir) [inline src/pages/jetton/util.ts:10]: restore hashes

File: README.md [hunk 10]
```diff
@@ -33,7 +33,8 @@ Open `http://localhost:3000`
 
 ## Is this tool safe?
 
-Yes. See https://github.com/ton-defi-org/jetton-deployer-contracts#qa-is-this-contract-deployer-safe
+Yes. See https://github.com/ton-blockchain/minter-contract
```
[the remark this answers is not in the document]
Comment (antongotech) [reply README.md:36]: fixed

File: src/App.tsx [hunk 11]
```diff
@@ -33,30 +44,35 @@ export const EnvContext = createContext({
   isTestnet: false,
 });
 
-function App() {
+const App = () => {
```
[the remark this answers is not in the document]
Comment (antongotech) [reply src/App.tsx:47]: func and const have no significant difference in terms of react components perfomance, it's just the modern and more style preferable way to create components

File: README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (antongotech) [reply README.md:36]: restored

File: src/components/header/headerSearchBar/SearchBar.tsx [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+export const SearchBar: React.FC<SearchBarProps> = ({ example, resetExample, closeMenu }) => {
+  const [value, setValue] = useState("");
+  const [active, setActive] = useState(false);
+  const [searchResults, setSearchResults] = useState<SearchRequest[]>([]);
+
+  const navigate = useNavigate();
+  const { showNotification } = useNotification();
+
+  const onSubmit = useCallback(async () => {
+    const searchResults = JSON.parse(window.localStorage.getItem(SEARCH_HISTORY) || "[]");
```
[the remark this answers is not in the document]
Comment (antongotech) [reply src/components/header/headerSearchBar/SearchBar.tsx:40]: fixed

File: src/components/header/headerSearchBar/SearchBar.tsx [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+      return item.value === value;
+    });
+
+    if (!value) {
+      return;
+    }
+
+    if (!isValidAddress(value)) {
+      showNotification("Invalid jetton address", "error");
+      setValue("");
```
[the remark this answers is not in the document]
Comment (antongotech) [reply src/components/header/headerSearchBar/SearchBar.tsx:51]: fixed

File: src/components/header/headerSearchBar/SearchBar.tsx [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+                    <img src={close} alt="Close Icon" width={18} height={18} />
+                  </IconButton>
+                </SearchResultsItem>
+              ))}
+              <CenteringWrapper mt={2} mb={1} ml={1} sx={{ width: "fit-content" }}>
+                <AppButton onClick={onHistoryClear} height={34} transparent>
+                  Clear History
+                </AppButton>
+              </CenteringWrapper>
+            </SearchResultsWrapper>
```
[the remark this answers is not in the document]
Comment (antongotech) [reply src/components/header/headerSearchBar/SearchBar.tsx:152]: separated

File: src/pages/deployer/data.ts (hunk 1 above)
Comment (antongotech) [reply src/pages/deployer/data.ts:52]: The text was modified according to the text in Figma template

File: src/pages/deployer/index.tsx (hunk 2 above)
Comment (antongotech) [reply src/pages/deployer/index.tsx:220]: restored

File: src/pages/deployer/index.tsx (hunk 3 above)
Comment (antongotech) [reply src/pages/deployer/index.tsx:240]: restored

File: src/pages/deployer/index.tsx (hunk 4 above)
Comment (antongotech) [reply src/pages/deployer/index.tsx:243]: Are you sure? Text was modified according to the Figma template

File: src/pages/deployer/index.tsx (hunk 6 above)
Comment (antongotech) [reply src/pages/deployer/index.tsx:163]: done

File: src/pages/jetton/actions/transfer/utils.tsx (hunk 8 above)
Comment (antongotech) [reply src/pages/jetton/actions/transfer/utils.tsx:5]: done

File: src/pages/deployer/index.tsx (hunk 7 above)
Comment (antongotech) [reply src/pages/deployer/index.tsx:278]: done

File: src/pages/jetton/util.ts (hunk 9 above)
Comment (antongotech) [reply src/pages/jetton/util.ts:10]: restored

File: src/components/header/headerSearchBar/SearchBar.tsx [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+    };
+  }, [value, onSubmit]);
+
+  useEffect(() => {
+    example && setValue(example);
+  }, [example]);
+
+  useEffect(() => {
+    window.localStorage.setItem(SEARCH_HISTORY, JSON.stringify(searchResults));
+  }, [searchResults]);
```
[the remark this answers is not in the document]
Comment (shaharyakir) [reply src/components/header/headerSearchBar/SearchBar.tsx:94]: actually that link is for a code snippet. what I mean is that this is generic logic that probably shouldn't be within a component.

File: src/pages/deployer/index.tsx (hunk 4 above)
Comment (shaharyakir) [reply src/pages/deployer/index.tsx:209]: For the moment, yes - figma usually shouldn't reflect product/copy decisions. I might be wrong on this specific one but I'll verify this alongside some other new texts

Comment (shaharyakir) [reply src/pages/deployer/index.tsx:209]: Same

File: src/pages/deployer/data.ts (hunk 1 above)
Comment (shaharyakir) [reply src/pages/deployer/data.ts:52]: similar - let's restore pls

File: src/App.tsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (shaharyakir) [reply src/App.tsx:47]: Okay, in this case it doesn't matter much, but for the contract verifier project let's not refactor it and keep the same style (using function).

File: src/pages/deployer/index.tsx (hunk 4 above)
Comment (antongotech) [reply src/pages/deployer/index.tsx:209]: restored

Comment (antongotech) [reply src/pages/deployer/index.tsx:209]: restored

**Pull request merged**